pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1240,7 +1241,8 @@ private SchemaData getSchema(Schema protocolSchema) {
             .isDeleted(false)
             .timestamp(System.currentTimeMillis())
             .user(Strings.nullToEmpty(originalPrincipal))
-            .type(Commands.getSchemaType(protocolSchema.getType()))
+            .type(protocolSchema.getType() == Schema.Type.AutoConsume
+                    ? SchemaType.AUTO_CONSUME : Commands.getSchemaType(protocolSchema.getType()))
```

Comment:
In addition to my comment before, we can avoid this change if you just modify the `Commands.getSchemaType` method.